runner.py

```diff
@@ -0,0 +1,707 @@
+        for test_config in config['tests']:
+            # Test consist of one pair (key, value) which is (name, config)
+            if len(test_config.keys()) != 1:
+                raise YAMLParserError(self, "test name not found in test config")
+
+            name = list(test_config.keys())[0]
+            test_config = test_config[name]
+
+            # Add useful staff to config
+            test_path = remove_prefix(str(self.path), str(PRTOS_PROJECT_DIR) + '/')
+            name = f"{test_path}/{name}"
+            test_config['name'] = name.replace('/', '.')
+            test_config['path'] = self.path
+
+            # Combine 'local' and 'global' config
+            for key in config:
+                if key not in test_config and key != 'tests':
+                    test_config[key] = config[key]
+
+            test = self.parse_test_case(test_config)
+            if test:
+                self.parsed_tests.append(test)
+
+        return self.parsed_tests
+
+
+class TestsRunner:
+    def __init__(self, targets, test_paths=None, build=True):
+        self.test_configs = []
+        self.test_paths = test_paths
+        self.tests_per_target = {k: [] for k in targets}
+        self.build = build
+        self.runners = {target: QemuRunner(QEMU_CMD[target]) for target in targets}
+        self.passed = 0
+        self.ignored = 0
+        self.failed = 0
+
+    def search_for_tests(self):
+        if not self.test_paths:
+            self.test_paths = list(PRTOS_TEST_DIR.rglob('test*.yaml'))
+            return
+
+        paths = []
+        for path in self.test_paths:
+            candidate = list(path.rglob('test*.yaml'))
+            if not candidate:
+                raise ValueError(f'Test {path} does not contain .yaml test configuration')
+            paths.extend(candidate)
+
+        self.test_paths = paths
+
+    def parse_tests(self):
+        self.test_configs = []
+        for path in self.test_paths:
+            parser = YAMLParser(path)
+            testcase_config = parser.parse_test_config()
+            self.test_configs.extend(testcase_config)
+            logging.debug(f"File {path} parsed successfuly")
+
+    def run(self):
+        self.search_for_tests()
+        self.parse_tests()
+
+        for test_config in self.test_configs:
+            for target, tests in self.tests_per_target.items():
+                test_config['target'] = target
+                test_case = TestCaseFactory.create(test_config)
+                tests.append(test_case)
+
+        if self.build:
+            for target, tests in self.tests_per_target.items():
+                target_builder = TargetBuilderFactory.create(target)
+
+                syspage = [test_case.syspage for test_case in tests]
+                syspage = list(set(syspage))
+                target_builder.build(syspage)
+
+        for target, tests in self.tests_per_target.items():
+            for test_case in tests:
+                self.runners[target].run(test_case)
```

Comment:
The test case name should be printed before the actual result, so I know what I'm waiting for (`print_test_name(); run(); print_result();`)

Reply:
I added `log_test_sart` and `log_test_status`.